src/jdk.net/aix/classes/jdk/net/AIXSocketOptions.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.net;
+
+import java.net.SocketException;
+import java.nio.file.attribute.UserPrincipal;
+import java.nio.file.attribute.GroupPrincipal;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import jdk.net.ExtendedSocketOptions.PlatformSocketOptions;
+import sun.nio.fs.UnixUserPrincipals;
+
+@SuppressWarnings("removal")
+class AIXSocketOptions extends PlatformSocketOptions {
+
+    public AIXSocketOptions() {
```

Comment:
It would be good to make this package private constructor instead of `public`, but I see that the current existing `LinuxSocketOptions` class too uses `public`.

Reply:
`newInstance("jdk.net.AIXSocketOptions")` will try to invoke the default constructor  `AIXSocketOptions() {}`. When we make it as private constructor PlatformSocketOptions instance will try to find jdk.net.AIXSocketOptions and within the time span it won't be able to find hence throws error (`Cannot get VM for test: java.net.SocketTimeoutException: Accept timed out`)

Reply:
Hello Varada, 
I went back and looked at how this class instance gets instantiated. The code uses reflection to do it:

```
private static PlatformSocketOptions newInstance(String cn) {
  Class<PlatformSocketOptions> c;
  try {
      c = (Class<PlatformSocketOptions>)Class.forName(cn);
      return c.getConstructor(new Class<?>[] { }).newInstance();
```
so yes, this expects a `public` constructor and that explains why the existing `LinuxSocketOptions` has a public constructor.

What you have right now, in this class, looks fine to me then.

Reply:
Thank you @jaikiran for the review